worker/apicaller/connect.go

```diff
@@ -83,6 +84,8 @@ func connectFallback(
 	conn api.Connection, didFallback bool, err error,
 ) {
 
+	// Encourage load balancing by defaulting to a different controller each time.
+	rand.Shuffle(len(info.Addrs), func(i, j int) { info.Addrs[i], info.Addrs[j] = info.Addrs[j], info.Addrs[i] })
```

Comment:
Unless I'm missing something, it seems like it would be better to do this in the code that actually handles the parallel connections (`dialWebsocketMulti` in `api/apiclient.go`), so that all clients that connect to the API server randomise the controller order (for example, with this diff `juju status` will continue to hit the first controller in the list).

Reply:
I wasn't sure about doing all clients, but some discussions convinced me that it should be reasonable. I do think there were reasons why we were intentionally not doing so, but I think that was more when we expected average latency to each controller being different, which is not actually likely to be true.